plugins/cicd-statistics/api-report.md

```diff
@@ -0,0 +1,182 @@
+## API Report File for "@backstage/plugin-cicd-statistics"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+/// <reference types="react" />
+
+import { ApiRef } from '@backstage/core-plugin-api';
+import { BackstagePlugin } from '@backstage/core-plugin-api';
+import { Entity } from '@backstage/catalog-model';
+import { RouteRef } from '@backstage/core-plugin-api';
+
+// Warning: (ae-missing-release-tag) "AbortError" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public
+export class AbortError extends Error {}
+
+// Warning: (ae-missing-release-tag) "branchTypes" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public (undocumented)
+export const branchTypes: Array<FilterBranchType>;
+
+// Warning: (ae-missing-release-tag) "Build" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public
+export interface Build {
+  buildType: FilterBranchType;
+  duration: number;
+  id: string;
+  // (undocumented)
+  raw?: unknown;
+  requestedAt: Date;
+  stages: Array<Stage>;
+  status: FilterStatusType;
+}
+
+// Warning: (ae-missing-release-tag) "BuildWithRaw" is exported by the package, but it is missing a release tag (@alpha, @beta, @public, or @internal)
+//
+// @public
+export type BuildWithRaw<T = any> = Build & {
```

Comment:
Do you think it would make sense to drop this and go for `Build<TRaw = unknown>` instead?

Reply:
So, we could actually remove this, and let implementors use something like it themselves. It's very useful though when constructing a list of these Builds and referring to the internal blobs you probably have in your particular implementation.

However, you shouldn't _have_ to use it (to e.g. set it to `undefined` or some other dummy), but if you **do** use it, you don't want it to be optional (`?`) so that you always need to `!` it. 🤔

So, **if** you want to use it in your API implementation, you can use `BuildWithRaw` with your own `MyBuildTypeThing` and return that back to this plugin, since it's compatible with `Build`. That was kind of the thinking here, although perhaps not very clear.